src/swap/SwapScreenV2.tsx

```diff
@@ -0,0 +1,1149 @@
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  const allowCrossChainSwaps = getFeatureGate(StatsigFeatureGates.ALLOW_CROSS_CHAIN_SWAPS)
+  const showUKCompliantVariant = getFeatureGate(StatsigFeatureGates.SHOW_UK_COMPLIANT_VARIANT)
+  const { swappableFromTokens, swappableToTokens, areSwapTokensShuffled } = useSwappableTokens()
+  const { links } = getDynamicConfigParams(DynamicConfigs[StatsigDynamicConfigs.APP_CONFIG])
+  const { maxSlippagePercentage, enableAppFee } = getDynamicConfigParams(
+    DynamicConfigs[StatsigDynamicConfigs.SWAP_CONFIG]
+  )
+
+  const inputFromRef = useRef<RNTextInput>(null)
+  const inputToRef = useRef<RNTextInput>(null)
+  const tokenBottomSheetFromRef = useRef<BottomSheetModalRefType>(null)
+  const tokenBottomSheetToRef = useRef<BottomSheetModalRefType>(null)
+  const exchangeRateInfoBottomSheetRef = useRef<BottomSheetModalRefType>(null)
+  const feeInfoBottomSheetRef = useRef<BottomSheetModalRefType>(null)
+  const slippageInfoBottomSheetRef = useRef<BottomSheetModalRefType>(null)
+  const estimatedDurationBottomSheetRef = useRef<BottomSheetModalRefType>(null)
+
+  const [noUsdPriceToken, setNoUsdPriceToken] = useState<
+    { token: TokenBalance; tokenPositionInList: number } | undefined
+  >(undefined)
+  const [selectedPercentage, setSelectedPercentage] = useState<number | null>(null)
+  const [startedSwapId, setStartedSwapId] = useState<string | undefined>(undefined)
+  const [switchedToNetworkId, setSwitchedToNetworkId] = useState<{
+    networkId: NetworkId
+    field: Field
+  } | null>(null)
+  const [fromToken, setFromToken] = useState(() => {
+    if (!route.params?.fromTokenId) return undefined
+    return swappableFromTokens.find((token) => token.tokenId === route.params!.fromTokenId)
+  })
+  const [toToken, setToToken] = useState(() => {
+    if (!route.params?.toTokenId) return undefined
+    return swappableToTokens.find((token) => token.tokenId === route.params!.toTokenId)
+  })
+
+  const currentSwap = useSelector(currentSwapSelector)
+  const localCurrency = useSelector(getLocalCurrencyCode)
+  const priceImpactWarningThreshold = useSelector(priceImpactWarningThresholdSelector)
+  const tokensById = useSelector((state) =>
+    tokensByIdSelector(state, getSupportedNetworkIdsForSwap())
+  )
+  const crossChainFeeCurrency = useSelector((state) =>
+    feeCurrenciesSelector(state, fromToken?.networkId || networkConfig.defaultNetworkId)
+  ).find((token) => token.isNative)
+  const feeCurrenciesWithPositiveBalances = useSelector((state) =>
+    feeCurrenciesWithPositiveBalancesSelector(
+      state,
+      fromToken?.networkId || networkConfig.defaultNetworkId
+    )
+  )
+
+  const { quote, refreshQuote, fetchSwapQuoteError, fetchingSwapQuote, clearQuote } = useSwapQuote({
+    networkId: fromToken?.networkId || networkConfig.defaultNetworkId,
+    slippagePercentage: maxSlippagePercentage,
+    enableAppFee: enableAppFee,
+    onError: (error) => {
```

Comment:
`onError` and `onSuccess` are the same as identical `useEffect`s from `SwapScreen.tsx`